### DISCUSSION TITLE
Why doesn't Pytest ignore warnings?

### DISCUSSION THREAD

**Discussion by cool-RR** [pytest-dev/pytest#10336] [General]
When I run tests, I'm getting DeprecationWarnings that are reported by Pytest. I would like Pytest to ignore these warnings and not report them.

I've put this in my `pyproject.toml`:

```
[tool.pytest.ini_options]
filterwarnings = [
    #'ignore:The distutils package is deprecated:DeprecationWarning',
    'ignore::DeprecationWarning',
]
```

However, I'm still seeing the warnings:

```
============================================================================================= warnings summary =============================================================================================
../../home/ramrachum/.venvs/ray_env/lib/python3.10/site-packages/gym/wrappers/monitoring/video_recorder.py:9
  /home/ramrachum/.venvs/ray_env/lib/python3.10/site-packages/gym/wrappers/monitoring/video_recorder.py:9: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Us
e setuptools or check PEP 632 for potential alternatives
    import distutils.spawn

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================================= 2 passed, 1 warning in 4.48s =======================================================================================
```

I run Pytest in the folder where the `pyproject.toml` file is. I'm sure that Pytest is reading the file, because when I put the wrong type quote on it, `pytest` complained about the error and wouldn't even run.

I also tried just putting a simple `'ignore'` in that configuration file, and also later a simple `'error'` but Pytest still produces the same output, showing the warning as a warning. 

What's wrong here? **How can I get Pytest to ignore these warnings?**

**Comment by nicoddemus** [comment]
Hi @cool-RR,

Indeed everything seems to be working OK. I know that depending on where the warning is produced pytest might not be able to ignore it yet because the configuration has not been loaded, but I don't remember the details.

If you want to investigate further, it would help to know at which point that warning is being captured by pytest. One simple way to do it is to edit (temporarily, just to test this) `/home/ramrachum/.venvs/ray_env/lib/python3.10/site-packages/gym/wrappers/monitoring/video_recorder.py:9` and add a `assert 0` right before the `import distutils.spawn`. This will blow up and show a traceback, and we can use it to see at which point we are in the pytest runtime.

**Reply by cool-RR** [reply]
Thanks @nicoddemus . Here's the output: https://gist.github.com/cool-RR/3ceced114249f6b74e58b81fdc85b7db

**Reply by nicoddemus** [reply]
Sorry, can you run again with `--full-trace`?

**Reply by cool-RR** [reply]
(Answered on the other answer, sorry for this.)

**Comment by cool-RR** [comment]
Here you go:
https://gist.github.com/cool-RR/a5412cad14d06b7f24675e0e9aa8e347

**Reply by nicoddemus** [reply]
Thanks!

Unfortunately I cannot tell why that particular warning is not being ignored... the warning is being triggered during a call to `import_module` inside `pytest_make_collect_report`, which happens inside `pytest_collection`. During the entire `pytest_collection` we do catch warnings, and apply the configured filters to them.

Any chance your codebase is changing the filters itself? Search for `filterwarnings` and see if you catch anything.

**Reply by cool-RR** [reply]
I searched the project, but the only place I found `filterwarnings` is in the snippet I included in my original message.

**Reply by nicoddemus** [reply]
Yeah no idea, I'm afraid... in fact I had used the same configuration to ignore that particular warning (`distutils.spawn`) in many projects already, and it does work. 😕

**Reply by cool-RR** [reply]
Oh well. Thanks for trying.
